doc/docusaurus/docs/using-plutus-tx/running-debugging.md

```diff
@@ -0,0 +1,328 @@
+If there is no CPU and/or MEM limit given, the budget is practically unlimited.
+
+``` shell
+$ echo "(program 1.1.0 (con integer 1))" | $plutus inc.plc --stdin --run --budget=229307,903
+Program passed all checks. No output file was written, use -o or --stdout.
+An error has occurred:
+The machine terminated part way through evaluation due to overspending the budget.
+The budget when the machine terminated was:
+({cpu: -1
+| mem: 1})
+Negative numbers indicate the overspent budget; note that this only indicates the budget that was needed for the next step, not to run the program to completion.
+$ echo "(program 1.1.0 (con integer 1))" | $plutus inc.plc --stdin --run --budget=,903
+Program passed all checks. No output file was written, use -o or --stdout.
+Execution succeeded. Final Term:
+(con integer 2)
+Remaining budget: ExBudget {exBudgetCPU = ExCPU 9223372036854546499, exBudgetMemory = ExMemory 1}
+Used budget: ExBudget {exBudgetCPU = ExCPU 229308, exBudgetMemory = ExMemory 902}
+```
+
+The executable takes as input-arguments the paths containing the plutus code to run. For example, consider the plutus code
+that expects two integers and adds them together:
+
+``` shell
+$ echo "(program 1.1.0 (lam x (lam y [(builtin subtractInteger) x y])))" > add.uplc
+$ plutus add.uplc
+Program passed all static checks. No output file was written, use -o or --stdout.
+```
+
+Same as above but using instead the `--stdin` option:
+
+``` shell
+$ echo "(program 1.1.0 (lam x (lam y [(builtin subtractInteger) x y])))" | plutus --stdin
+Program passed all static checks. No output file was written, use -o or --stdout.
+```
+
+Fixing the syntax error above makes the evaluator succeed, returning the final value
+as the result of the evaluation (a function in our example):
+
+``` shell
+$ echo "(program 1.1.0 (lam x (lam y [(builtin subtractInteger) x y])))" | plutus --stdin --stdout
+(program 1.1.0 (lam x-0 (lam y-1 [ [ (builtin subtractInteger) x-0 ] y-1 ])))
+```
+
+note :: pin FIXME
+
+As shown in [compilation](FIXME) section, you are free to mix and match
+different input languages (pir/tplc/plc); as long as the output target at the CLI
+is set to `plc`, their compiled output will be `--run` through the
+the local `plc` interpreter (same interpreter as on-chain). Alternatively, you can
+target `tplc` which changes to a different, `tplc` interpreter. Although
+the `tplc` interpreter behaves the same as the default `plc` interpreter for *type correct* programs,
+it is not advised to use it because it comes with caveats: cannot execute `plc` code,
+cannot have budget accounting and budget limits, runs way slower and your program must be fully type correct.
+The last point is not necessarily a caveat, but it diverges from the on-chain behaviour:
+the `tplc` interpreter accepts less programs than the default `plc` interpreter.
```

Comment:
```suggestion
different input languages (pir/tplc/uplc); as long as the output target at the CLI
is set to `uplc`, their compiled output will be `--run` through the
the local `uplc` interpreter (same interpreter as on-chain). Alternatively, you can
target `tplc` which changes to a different, `tplc` interpreter. Although
the `tplc` interpreter behaves the same as the default `uplc` interpreter for *type correct* programs,
it is not advised to use it because it comes with caveats: cannot execute `uplc` code,
cannot have budget accounting and budget limits, runs way slower and your program must be fully type correct.
The last point is not necessarily a caveat, but it diverges from the on-chain behaviour:
the `tplc` interpreter accepts less programs than the default `uplc` interpreter.
```